shared-libs/lineage/src/hydration.js

```diff
@@ -328,7 +325,8 @@ module.exports = function(Promise, DB) {
       })
       .then(function(secondRoundFetched) {
         knownDocs.push(...secondRoundFetched);
-        
+
+        fillContactsInDocs(knownDocs, knownDocs);
```

Comment:
so you hydrate each doc in knownDocs with all known data in knownDocs. i am a little unclear why this is required ... seems it could make many deep copies. This code path is shared by `fetchLineageByIds` - did you check how performance of that path is impacted?

Reply:
So, your `knownDocs` are the parents and the patients. The way the code worked before, this was called as part of `mergeLineagesIntoDoc` for subsets of "knownDocs", and at least once for every item in "knownDocs" - because every item from "knownDocs" would be part of some compound lineage. 
`fillContactsInDocs` iterates over the first parameter and fills the `.contact` property with docs from the 2nd parameter. So calling it for the whole collection ONCE compared to calling it for subsets of the collection multiple times is arguably more performant, because if you pass over the "lineage" twice, you just hydrate it once, instead of repeated times, which was the problem in the 1st place.

Reply:
I don't exactly understand this comment: "seems it could make many deep copies". 

`fillContactsInDocs` is going to hydrate the ".contact" property with the actual contact doc for every doc in our collection. How would this make too many deep copies? This was already happening but with subsets of "knownDocs". 
Do you think it's not correct to hydrate the ".contact" property of every lineage doc we're working with?